apps/server/src/modules/h5p-editor/service/temporary-file-storage.service.ts

```diff
@@ -1,79 +1,110 @@
-import { ITemporaryFile, ITemporaryFileStorage, IUser } from '@lumieducation/h5p-server';
-import { Inject, Injectable, NotAcceptableException } from '@nestjs/common';
 import { S3ClientAdapter } from '@infra/s3-client';
+import { IFileStats, ITemporaryFile, ITemporaryFileStorage, IUser } from '@lumieducation/h5p-server';
+import {
+	HttpException,
+	Inject,
+	Injectable,
+	InternalServerErrorException,
+	NotAcceptableException,
+	NotFoundException,
+} from '@nestjs/common';
+import { ErrorUtils } from '@src/core/error/utils';
 import { ReadStream } from 'fs';
 import { Readable } from 'stream';
 import { H5pFileDto } from '../controller/dto/h5p-file.dto';
-import { H5pEditorTempFile } from '../entity/h5p-editor-tempfile.entity';
 import { H5P_CONTENT_S3_CONNECTION } from '../h5p-editor.config';
-import { TemporaryFileRepo } from '../repo/temporary-file.repo';
 
 @Injectable()
 export class TemporaryFileStorage implements ITemporaryFileStorage {
-	constructor(
-		private readonly repo: TemporaryFileRepo,
-		@Inject(H5P_CONTENT_S3_CONNECTION) private readonly s3Client: S3ClientAdapter
-	) {}
+	constructor(@Inject(H5P_CONTENT_S3_CONNECTION) private readonly s3Client: S3ClientAdapter) {}
 
 	private checkFilename(filename: string): void {
-		if (!/^[a-zA-Z0-9/._-]+$/g.test(filename) && filename.includes('..') && filename.startsWith('/')) {
-			throw new NotAcceptableException(`Filename contains forbidden characters or is empty: '${filename}'`);
+		filename = filename.split('.').slice(0, -1).join('.');
+		if (/^[a-zA-Z0-9/._-]*$/.test(filename) && !filename.includes('..') && !filename.startsWith('/')) {
+			return;
 		}
-	}
-
-	private getFileInfo(filename: string, userId: string): Promise<H5pEditorTempFile> {
-		this.checkFilename(filename);
-		return this.repo.findByUserAndFilename(userId, filename);
+		throw new HttpException('message', 406, {
+			cause: new NotAcceptableException(`Filename contains forbidden characters ${filename}`),
+		});
 	}
 
 	public async deleteFile(filename: string, userId: string): Promise<void> {
 		this.checkFilename(filename);
-		const meta = await this.repo.findByUserAndFilename(userId, filename);
-		await this.s3Client.delete([this.getFilePath(userId, filename)]);
-		await this.repo.delete(meta);
+		const filePath = this.getFilePath(userId, filename);
+		await this.s3Client.delete([filePath]);
 	}
 
 	public async fileExists(filename: string, user: IUser): Promise<boolean> {
 		this.checkFilename(filename);
-		const files = await this.repo.findAllByUserAndFilename(user.id, filename);
-		const exists = files.length !== 0;
-		return exists;
+
+		const filePath = this.getFilePath(user.id, filename);
+
+		try {
+			await this.s3Client.get(filePath);
```

Comment:
I think you should use the head command

https://github.com/hpi-schul-cloud/schulcloud-server/blob/main/apps/server/src/infra/s3-client/s3-client.adapter.ts#L243

https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadObject.html

Reply:
If I recall correctly, there were weird race-conditions when using the HEAD-command.
Lumi calls both the `fileExists` and `getFileStream` methods at the same time and depending on the order of execution the stream might just stop. 
I think that was the reason why we switched over to GET in the ContentStorage, which is where I took this implementation from.